socorro/processor/cache_manager.py

```diff
@@ -293,15 +307,45 @@ def _event_generator(self, nonblocking=False):
                             # Handle directory events which update our watch lists
                             if flags.CREATE & event_mask:
                                 try:
-                                    self.add_watch(path)
+                                    created_wd = self.add_watch(path)
                                 except OSError:
+                                    logger.exception("add watch error for %s", path)
                                     continue
 
-                                # This is a new directory to watch, so we should add the
-                                # contents to our LRU and watches
-                                self.inventory_existing(path)
+                                # This is a new directory to watch, so we add faked
+                                # events for subdirectories and files
+                                for name in os.listdir(path):
```

Comment:
We do a shallow directory listing because we're going to add events that, when handled, will cover sub directories.